src/components/common/OfficeAddresses.vue

```diff
@@ -278,185 +270,135 @@ export default class OfficeAddresses extends Mixins(CommonMixin) {
+  @Watch('asOfDate', { immediate: true })
+  private async onAsOfDateChanged (date: string): Promise<void> {
+    console.log('*** got asOfDate from parent =', date)
 
-      // if the address has changed from the original, set action flag
-      this.addressModified(this.recDeliveryAddress, this.recDeliveryAddressOriginal)
-        ? this.addAction(recDeliveryAddressFinal, Actions.ADDRESSCHANGED)
-        : this.removeAction(recDeliveryAddressFinal, Actions.ADDRESSCHANGED)
+    // fetch original addresses
+    await this.fetchAddresses()
 
-      // if the address has changed from the original, set action flag
-      this.addressModified(this.recMailingAddress, this.recMailingAddressOriginal)
-        ? this.addAction(recMailingAddressFinal, Actions.ADDRESSCHANGED)
-        : this.removeAction(recMailingAddressFinal, Actions.ADDRESSCHANGED)
-    }
+    // fill current addresses from original
+    // this may be overwritten by draft data
+    // (assumes Addresses changes after AsOfDate)
+    this.assignAddresses(this.original)
 
-    if (this.recordsAddress) {
-      return {
-        registeredOffice: {
-          deliveryAddress: deliveryAddressFinal,
-          mailingAddress: mailingAddressFinal
-        },
-        recordsOffice: {
-          deliveryAddress: recDeliveryAddressFinal,
-          mailingAddress: recMailingAddressFinal
-        }
-      }
-    } else {
-      return {
-        registeredOffice: {
-          deliveryAddress: deliveryAddressFinal,
-          mailingAddress: mailingAddressFinal
-        }
-      }
-    }
+    console.log('*** onAsOfDateChanged() emitting addresses...')
+    this.emitAddresses()
+    this.emitModified()
   }
 
-  /**
-   * Called when addresses property changes (ie, when parent page has loaded a draft filing).
-   */
+  /** Called when Addresses prop changes (ie, when parent has loaded a draft filing). */
   @Watch('addresses')
-  onAddressesChanged (): void {
-    if (this.addresses && this.addresses.recordsOffice) {
-      this.inheritRegisteredAddress =
-        this.isSame(this.addresses.registeredOffice.deliveryAddress,
-          this.addresses.recordsOffice.deliveryAddress, 'actions') &&
-        this.isSame(this.addresses.registeredOffice.mailingAddress,
-          this.addresses.recordsOffice.mailingAddress, 'actions')
-    }
-    this.deliveryAddress = isEmpty(this.addresses)
-      ? {} as AddressIF
-      : this.addresses.registeredOffice['deliveryAddress']
-    this.mailingAddress = isEmpty(this.addresses)
-      ? {} as AddressIF
-      : this.addresses.registeredOffice['mailingAddress']
-
-    if (this.recordsAddress) {
-      this.recDeliveryAddress = isEmpty(this.addresses)
-        ? {} as AddressIF
-        : this.addresses.recordsOffice['deliveryAddress']
-
-      this.recMailingAddress = isEmpty(this.addresses)
-        ? {} as AddressIF
-        : this.addresses.recordsOffice['mailingAddress']
-    }
+  private onAddressesChanged (): void {
```

Comment:
FYI, since "addresses" is a prop, when we emit an update event for it (ie, sync modifier in parent) then we receive it back here.

To me it feels weird to "hear back from the parent that we updated its addresses", but then we (component) can't update the prop directly anyway, so I guess this is how it goes.